modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/runtime/resolver/NestedProcessorValueResolverTestCase.java

```diff
@@ -24,40 +25,54 @@
 import org.mockito.ArgumentCaptor;
 import org.mockito.Mock;
 import org.mockito.runners.MockitoJUnitRunner;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Mono;
 
 @RunWith(MockitoJUnitRunner.class)
 public class NestedProcessorValueResolverTestCase extends AbstractMuleContextTestCase {
 
   @Mock
-  private Processor messageProcessor;
+  private MessageProcessorChain messageProcessor;
 
   @Before
   public void before() throws Exception {
     final InternalEvent testEvent = testEvent();
     when(messageProcessor.process(any(InternalEvent.class))).thenReturn(testEvent);
+    when(messageProcessor.apply(any(Publisher.class))).thenReturn(Mono.just(testEvent));
   }
 
   @Test
   public void yieldsNestedProcessor() throws Exception {
-    NestedProcessorValueResolver resolver = new NestedProcessorValueResolver(messageProcessor);
-    resolver.setMuleContext(muleContext);
-    NestedProcessor nestedProcessor = resolver.resolve(ValueResolvingContext.from(testEvent()));
-    Object response = nestedProcessor.process();
-    assertThat(response, is(TEST_PAYLOAD));
+    ProcessorChainValueResolver resolver = new ProcessorChainValueResolver(messageProcessor);
+    muleContext.getInjector().inject(resolver);
+    final InternalEvent event = testEvent();
 
-    ArgumentCaptor<InternalEvent> captor = ArgumentCaptor.forClass(InternalEvent.class);
-    verify(messageProcessor).process(captor.capture());
+    Chain nestedProcessor = resolver.resolve(ValueResolvingContext.from(event));
+    nestedProcessor
+        .onSuccess(result -> {
+          assertThat(result.getOutput(), is(TEST_PAYLOAD));
 
-    InternalEvent capturedEvent = captor.getValue();
-    assertThat(capturedEvent, is(testEvent()));
+          ArgumentCaptor<InternalEvent> captor = ArgumentCaptor.forClass(InternalEvent.class);
+          try {
+            verify(messageProcessor).process(captor.capture());
+          } catch (MuleException e) {
+            throw new RuntimeException(e);
+          }
+
+          InternalEvent capturedEvent = captor.getValue();
+          assertThat(capturedEvent, is(event));
+        })
+        .onError((e, r) -> {
+          throw new RuntimeException(e);
+        });
   }
 
   @Test
   public void alwaysGivesDifferentInstances() throws Exception {
-    NestedProcessorValueResolver resolver = new NestedProcessorValueResolver(messageProcessor);
-    resolver.setMuleContext(muleContext);
-    NestedProcessor resolved1 = resolver.resolve(ValueResolvingContext.from(testEvent()));
-    NestedProcessor resolved2 = resolver.resolve(ValueResolvingContext.from(testEvent()));
+    ProcessorChainValueResolver resolver = new ProcessorChainValueResolver(messageProcessor);
+    muleContext.getInjector().inject(resolver);
+    Chain resolved1 = resolver.resolve(ValueResolvingContext.from(testEvent()));
```

Comment:
This is very wrong. It should always be the same chain. What happens if I put a stateful operation in the scope? This is not compatible with an idempotent filter for example or with <ee:cache>

Reply:
to me more clear, the MEssageprocessorChain must always be the same. But the SDK chain must always be a different instance

Reply:
see `mule/modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/ScopeExecutionTestCase#verifySameProcessorInstance`